wormhole-connect/src/views/v2/Redeem/index.tsx

```diff
@@ -164,30 +170,107 @@ const Redeem = () => {
+    amount,
+    getUSDAmount,
+  );
+
+  const { classes } = useStyles({ transitionDuration: `${eta}ms` });
+
+  // Handle changes to receiptState as well as uncaught errors when initiating manual redeems
+  // There are four cases this handles, in this order:
+  //
+  // - receipt.state === DestinationFinalized
+  // - receipt.state === Refunded
+  // - receipt.state === Failed
+  // - Unhandled error when manually redeeming
+  //
+  // Because the unhandled manual redeem error is at the end, we ignore it if
+  // we already saw one of the first three cases.
   useEffect(() => {
-    // When we see the transfer was complete for the first time,
-    // fire a transfer.success telemetry event.
-    if (isTxComplete && !transferSuccessEventFired) {
-      setTransferSuccessEventFired(true);
+    const { receipt } = routeContext;
+
+    if (!receipt) return;
+
+    if (isTxComplete) {
+      if (!transferSuccessEventFired) {
+        // When we see the transfer was complete for the first time,
+        // fire a transfer.success telemetry event.
+        setTransferSuccessEventFired(true);
+
+        config.triggerEvent({
+          type: 'transfer.success',
+          details,
+        });
+
+        if (!isAutomaticRoute) {
+          // Manual routes also fire a second success event specific to manual redeems
+          config.triggerEvent({
+            type: 'transfer.redeem.success',
+            details,
+          });
+        }
 
+        setIsClaimInProgress(false);
+        setClaimError('');
+      }
+    } else if (isTxRefunded) {
       config.triggerEvent({
-        type: 'transfer.success',
-        details: getTransferDetails(
-          routeName!,
-          tokenKey,
-          receivedTokenKey,
-          fromChain,
-          toChain,
-          amount,
-          getUSDAmount,
-        ),
+        type: 'transfer.refunded',
+        details,
       });
+    } else if (isFailed(receipt)) {
+      const [uiError, transferError] = interpretTransferError(
+        receipt.error,
+        details,
+      );
+      setClaimError(uiError);
+
+      config.triggerEvent({
+        type: 'transfer.error',
+        details,
+        error: transferError,
+      });
+
+      console.error(
+        `Transfer failed with error ${transferError}: ${receipt.error}`,
+      );
+
+      setIsClaimInProgress(false);
+    } else if (unhandledManualClaimError) {
+      const [uiError, transferError] = interpretTransferError(
+        unhandledManualClaimError,
+        details,
+      );
+
+      setClaimError(uiError);
+
+      config.triggerEvent({
+        type: 'transfer.redeem.error',
+        details,
+        error: transferError,
+      });
+
+      console.error(
+        `Error while manually redeeming: ${transferError.type} - ${unhandledManualClaimError}`,
+      );
+
+      setIsClaimInProgress(false);
     }
-  }, [isTxComplete]);
+  }, [
+    receipt?.state,
+    isTxComplete,
+    isTxRefunded,
+    unhandledManualClaimError,
```

Comment:
We should remove one of the `unhandledManualClaimError` and add `transferSuccessEventFired` and `isAutomaticRoute`

Reply:
wow oops

Reply:
we technically dont need `transferSuccessEventFired` in this because this hook is the only thing that manipulates it. but I'll add it anyway for strictness' sake